src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -592,41 +642,12 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
 
         if (auto result = basicBlocks.find(offset); result != basicBlocks.end())
         {
-            builder.SetInsertPoint(result->second);
-        }
-
-        auto addBasicBlock = [&](std::uint16_t target)
-        {
-            auto [result, inserted] = basicBlocks.insert({target, nullptr});
-            if (!inserted)
+            if (builder.GetInsertBlock()->getTerminator() == nullptr)
```

Comment:
Could you leave a comment above here maybe documenting as to why this is necassery? Since this was non-trivial